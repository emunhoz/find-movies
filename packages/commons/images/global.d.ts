declare module '@commons/images'
declare module '*.svg'
declare module '*.png'
declare module '*.jpg'
