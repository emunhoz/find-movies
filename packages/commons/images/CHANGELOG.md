# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.1.1](https://github.com/emunhoz/find-movies/compare/@commons/images@0.1.0...@commons/images@0.1.1) (2021-02-15)

**Note:** Version bump only for package @commons/images





# [0.1.0](https://github.com/emunhoz/find-movies/compare/@commons/images@0.0.2...@commons/images@0.1.0) (2021-02-15)


### Features

* add searchBar component ([72623a7](https://github.com/emunhoz/find-movies/commit/72623a78ad0a0f5ccc2fc9cde4aa53c9482336d5))





## 0.0.2 (2021-02-15)

**Note:** Version bump only for package @commons/images
