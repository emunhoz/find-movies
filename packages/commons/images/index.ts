import IconHeart from './icons/icon-heart.svg'
import IconHeartFull from './icons/icon-heart-full.svg'
import IconHeartWhite from './icons/icon-heart-white.svg'
import SearchIcon from './icons/icon-magnifier-white.svg'
import ArrowWhite from './icons/icon-arrow-white.svg'
import EmptyState from './illustrations/empty-state.png'
import ImageNotFound from './illustrations/image-not-found.svg'
import Logo from './logos/logo.svg'
import LogoImdb from './logos/logo-imdb.svg'
import LogoRottenTomatoes from './logos/logo-rotten-tomatoes.svg'
import LogoMetacritic from './logos/logo-metacritic.svg'

export {
  IconHeart,
  IconHeartFull,
  IconHeartWhite,
  SearchIcon,
  ArrowWhite,
  EmptyState,
  ImageNotFound,
  Logo,
  LogoImdb,
  LogoRottenTomatoes,
  LogoMetacritic
}
