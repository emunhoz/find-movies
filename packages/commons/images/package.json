{
  "name": "@commons/images",
  "version": "0.1.8",
  "main": "index.ts",
  "types": "global.d.ts",
  "license": "MIT"
}
