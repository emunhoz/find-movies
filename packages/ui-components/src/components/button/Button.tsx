import React from 'react'
import * as S from './Button.style'

interface IButton {
  children?: React.ReactNode
  icon?: string
  onClick?: () => void
  disabled?: boolean
  /**
   * Styles button as 100% width
   *
   * @default false
   */
  block?: boolean
  /**
   * Styles button with background active style
   *
   * @default false
   */
  active?: boolean
  /**
   * Ghost style
   *
   * @default false
   */
  ghost?: boolean
  /**
   * Defines the size of the button
   *  big: 66px
   *  small: 32px
   *
   * @default default
   */
  size?: 'big' | 'default' | 'small';
}

export const Button = ({ children, icon, disabled, size = 'default', ghost, onClick, block = false, active = false, ...rest }: IButton) => (
  <S.Button onClick={onClick} ghost={ghost} block={block} size={size} active={active} {...rest}>
    {icon && <img src={icon} />}
    {children}
  </S.Button>
)
