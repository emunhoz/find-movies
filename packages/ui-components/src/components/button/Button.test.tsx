import React from 'react'
import { render, fireEvent, screen } from '../../utils/test-utils'
import { IconHeartWhite } from '@commons/images'
import { Button } from './Button'

describe('[Component]: Button', () => {
  const mockOnClickFunction = jest.fn()

  test('render the component', () => {
    const { asFragment } = render(<Button>Hello world</Button>)

    expect(asFragment()).toMatchSnapshot()
  })

  test('render active style', () => {
    const { asFragment } = render(<Button active>Hello world</Button>)

    expect(asFragment()).toMatchSnapshot()
  })

  test('render block style', () => {
    const { asFragment } = render(<Button block>Hello world</Button>)

    expect(asFragment()).toMatchSnapshot()
  })

  test('should trigger function when button is pressed', () => {
    render(<Button onClick={mockOnClickFunction}>Hello world</Button>)

    fireEvent.click(screen.getByText('Hello world'))

    expect(mockOnClickFunction).toHaveBeenCalled()
  })

  test('render component disabled', () => {
    const { asFragment } = render(<Button disabled>Hello world</Button>)

    expect(asFragment()).toMatchSnapshot()
  })

  test('render component with icon', () => {
    const { asFragment } = render(<Button icon={IconHeartWhite}>Hello world</Button>)

    expect(asFragment()).toMatchSnapshot()
  })
})
