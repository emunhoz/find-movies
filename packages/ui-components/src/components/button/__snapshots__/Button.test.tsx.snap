// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component]: Button render component disabled 1`] = `
<DocumentFragment>
  <button
    class="sc-bdfBwQ kMxBQV"
    disabled=""
  >
    Hello world
  </button>
</DocumentFragment>
`;

exports[`[Component]: Button render component with big size style 1`] = `
<DocumentFragment>
  <button
    class="sc-bdfBwQ chhkiQ"
  >
    Hello world
  </button>
</DocumentFragment>
`;

exports[`[Component]: Button render component with small size style 1`] = `
<DocumentFragment>
  <button
    class="sc-bdfBwQ esBSOR"
  >
    Hello world
  </button>
</DocumentFragment>
`;

exports[`[Component]: Button render ghost style 1`] = `
<DocumentFragment>
  <button
    class="sc-bdfBwQ jxhIFl"
  >
    Hello world
  </button>
</DocumentFragment>
`;

exports[`[Component]: Button render negative style 1`] = `
<DocumentFragment>
  <button
    class="sc-bdfBwQ dCAyoC"
  >
    Hello world
  </button>
</DocumentFragment>
`;

exports[`[Component]: Button render the component 1`] = `
<DocumentFragment>
  <button
    class="sc-bdfBwQ kMxBQV"
  >
    Hello world
  </button>
</DocumentFragment>
`;
